--- Bonmin/src/IpoptInterface/CoinMessageHandler2Journal.cpp	(Revision 392)
+++ Bonmin/src/IpoptInterface/CoinMessageHandler2Journal.cpp	(Arbeitskopie)
@@ -31,7 +31,7 @@
 CoinMessageHandler2Journal::~CoinMessageHandler2Journal() { }
 
 /** Print to the designated output location */
-void CoinMessageHandler2Journal::PrintImpl(const char* str) {
+void CoinMessageHandler2Journal::PrintImpl(EJournalCategory category, EJournalLevel level, const char* str) {
   //TODO one should set the detail level of the message according to the journalist print level
  
   if (messagehandler) {
@@ -68,7 +68,7 @@
 }
 
 /** Printf to the designated output location */
-void CoinMessageHandler2Journal::PrintfImpl(const char* pformat, va_list ap) {
+void CoinMessageHandler2Journal::PrintfImpl(EJournalCategory category, EJournalLevel level, const char* pformat, va_list ap) {
   char outBuf[1024];
 
 #ifdef HAVE_VA_COPY
@@ -80,7 +80,7 @@
   vsnprintf(outBuf, sizeof(outBuf), pformat, ap);
 #endif
 
-		 PrintImpl(outBuf);
+		 PrintImpl(category, level, outBuf);
 }
 
 /** Flush output buffer.*/
--- Bonmin/src/IpoptInterface/CoinMessageHandler2Journal.hpp	(Revision 392)
+++ Bonmin/src/IpoptInterface/CoinMessageHandler2Journal.hpp	(Arbeitskopie)
@@ -48,10 +48,10 @@
 
 protected:
   /** Print to the designated output location */
-  virtual void PrintImpl(const char* str);
+  virtual void PrintImpl(EJournalCategory category, EJournalLevel level, const char* str);
 
   /** Printf to the designated output location */
-  virtual void PrintfImpl(const char* pformat, va_list ap);
+  virtual void PrintfImpl(EJournalCategory category, EJournalLevel level, const char* pformat, va_list ap);
 
   /** Flush output buffer.*/
   virtual void FlushBufferImpl();
--- Bonmin/src/IpoptInterface/IpoptInterface.hpp	(Revision 392)
+++ Bonmin/src/IpoptInterface/IpoptInterface.hpp	(Arbeitskopie)
@@ -51,8 +51,8 @@
   {
   public:
     /// Default constructor
-    SimpleError() : CoinError()
-    {}
+//    SimpleError() : CoinError()
+//    {}
 
     ///Alternate constructor using strings
     SimpleError(std::string message,
