Index: Bonmin/src/Interfaces/Ipopt/BonIpoptSolver.cpp
===================================================================
--- Bonmin/src/Interfaces/Ipopt/BonIpoptSolver.cpp	(Revision 1065)
+++ Bonmin/src/Interfaces/Ipopt/BonIpoptSolver.cpp	(Arbeitskopie)
@@ -245,7 +245,7 @@
     case Ipopt::Solve_Succeeded:
       return solvedOptimal;
     case Ipopt::Search_Direction_Becomes_Too_Small:
-      std::cerr<<"Warning : need to verify that Search_Direction_Becomes_Too_Small is indeed Ok"<<std::endl;
+//      std::cerr<<"Warning : need to verify that Search_Direction_Becomes_Too_Small is indeed Ok"<<std::endl;
     case Ipopt::Solved_To_Acceptable_Level:
       return solvedOptimalTol;
     case Ipopt::Infeasible_Problem_Detected:
