--- Cbc/src/CbcStrategy.cpp	(Revision 522)
+++ Cbc/src/CbcStrategy.cpp	(Arbeitskopie)
@@ -353,6 +353,7 @@
         OsiClpSolverInterface * clpSolver = dynamic_cast< OsiClpSolverInterface*> (solver);
         ClpSimplex * lpSolver = clpSolver->getModelPtr();
         if (lpSolver->tightenPrimalBounds()==0) {
+	        lpSolver->passInMessageHandler(solver->messageHandler());
           lpSolver->dual();
         } else {
           feasible = false;
--- Cbc/src/CbcBranchActual.cpp	(Revision 522)
+++ Cbc/src/CbcBranchActual.cpp	(Arbeitskopie)
@@ -232,7 +232,7 @@
     }
   }
   if (n2<numberMembers_) {
-    printf("** SOS number of members reduced from %d to %d!\n",numberMembers_,n2);
+//    printf("** SOS number of members reduced from %d to %d!\n",numberMembers_,n2);
     numberMembers_=n2;
   }
   // Find out how many non sos
@@ -538,7 +538,7 @@
     }
   }
   if (n2<numberMembers_) {
-    printf("** SOS number of members reduced from %d to %d!\n",numberMembers_,n2);
+//    printf("** SOS number of members reduced from %d to %d!\n",numberMembers_,n2);
     numberMembers_=n2;
   }
 }
