Index: src/CglPreProcess/CglPreProcess.cpp
===================================================================
--- src/CglPreProcess/CglPreProcess.cpp	(Revision 371)
+++ Cgl/src/CglPreProcess/CglPreProcess.cpp	(Arbeitskopie)
@@ -1283,8 +1283,8 @@
         }
       }
       if (numberChanges) {
-        if (handler_->logLevel()>1) 
-          printf("%d elements changed\n",numberChanges);
+//        if (handler_->logLevel()>1) 
+//          printf("%d elements changed\n",numberChanges);
         model.replaceMatrixOptional(copy);
       }
     }
