Index: CoinUtils/configure
===================================================================
--- CoinUtils/configure	(revision 1241)
+++ CoinUtils/configure	(working copy)
@@ -28570,6 +28570,7 @@
 
 if test x"$use_thirdpartyglpk" = xbuild; then
   COINUTILS_LIBADD="../../ThirdParty/Glpk/libcoinglpk.la $COINUTILS_LIBADD"
+  ADDLIBS="$ADDLIBS `cat ../ThirdParty/Glpk/glpk_addlibs.txt`"
 fi
 
 #############################################################################
Index: CoinUtils/configure.ac
===================================================================
--- CoinUtils/configure.ac	(revision 1241)
+++ CoinUtils/configure.ac	(working copy)
@@ -446,6 +446,7 @@
 AC_COIN_HAS_GLPK(Gmpl)
 if test x"$use_thirdpartyglpk" = xbuild; then
   COINUTILS_LIBADD="../../ThirdParty/Glpk/libcoinglpk.la $COINUTILS_LIBADD"
+  ADDLIBS="$ADDLIBS `cat ../ThirdParty/Glpk/glpk_addlibs.txt`"
 fi
 
 #############################################################################
Index: CoinUtils/src/Makefile.in
===================================================================
--- CoinUtils/src/Makefile.in	(revision 1241)
+++ CoinUtils/src/Makefile.in	(working copy)
@@ -43,7 +43,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@COIN_HAS_GMPL_TRUE@am__append_1 = -I`$(CYGPATH_W) $(GMPLINCDIR)`
+@COIN_HAS_GMPL_TRUE@am__append_1 = -I`$(CYGPATH_W) $(GMPLINCDIR)` -I`$(CYGPATH_W) $(GMPLOBJDIR)`
 subdir = src
 DIST_COMMON = $(includecoin_HEADERS) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in
Index: CoinUtils/src/Makefile.am
===================================================================
--- CoinUtils/src/Makefile.am	(revision 1241)
+++ CoinUtils/src/Makefile.am	(working copy)
@@ -112,7 +112,7 @@
 # "cygpath" stuff is necessary to compile with native compilers on Windows
 AM_CPPFLAGS =
 if COIN_HAS_GMPL
-  AM_CPPFLAGS += -I`$(CYGPATH_W) $(GMPLINCDIR)`
+  AM_CPPFLAGS += -I`$(CYGPATH_W) $(GMPLINCDIR)` -I`$(CYGPATH_W) $(GMPLOBJDIR)`
 endif
 
 # This line is necessary to allow VPATH compilation with MS compilers
