Index: Ipopt/src/Algorithm/IpRestoConvCheck.cpp
===================================================================
--- Ipopt/src/Algorithm/IpRestoConvCheck.cpp	(revision 1572)
+++ Ipopt/src/Algorithm/IpRestoConvCheck.cpp	(working copy)
@@ -160,9 +160,6 @@
     Number orig_inf_pr_max = Max(kappa_resto_*orig_curr_inf_pr,
                                  Min(orig_ip_data->tol(),
                                      orig_constr_viol_tol_));
-    if (kappa_resto_ == 0.) {
-      orig_inf_pr_max = 0.;
-    }
 
     if (first_resto_iter_) {
       Jnlst().Printf(J_DETAILED, J_MAIN,
