Index: Osi/src/OsiGlpk/OsiGlpkSolverInterface.cpp
===================================================================
--- Osi/src/OsiGlpk/OsiGlpkSolverInterface.cpp	(revision 1023)
+++ Osi/src/OsiGlpk/OsiGlpkSolverInterface.cpp	(working copy)
@@ -176,6 +176,7 @@
   If we ever reach the default case, we're deeply confused.
 */
   isIterationLimitReached_ = false ;
+  isTimeLimitReached_ = false;
   isAbandoned_ = false ;
   isPrimInfeasible_ = false ;
   isDualInfeasible_ = false ;
@@ -195,6 +196,8 @@
     { isObjUpperLimitReached_ = true ;
       break ; }
     case LPX_E_TMLIM:
+    { isTimeLimitReached_ = true ;
+    } // no break here, so we still report abandoned
     case LPX_E_FAULT:
     case LPX_E_SING:
     { isAbandoned_ = true ;
@@ -231,6 +234,7 @@
   iter_used_ = lpx_get_int_parm(model,LPX_K_ITCNT) ;
 
   isIterationLimitReached_ = false ;
+  isTimeLimitReached_ = false ;
   isAbandoned_ = false ;
   isObjLowerLimitReached_ = false ;
   isObjUpperLimitReached_ = false ;
@@ -250,6 +254,8 @@
     { isObjUpperLimitReached_ = true ;
       break ; }
     case LPX_E_TMLIM:
+    { isTimeLimitReached_ = true ;
+    } // no break here, so we still report abandoned
     case LPX_E_FAULT:
     case LPX_E_SING:
     { isAbandoned_ = true ;
@@ -340,6 +346,7 @@
 */
     iter_used_ = lpx_get_int_parm(model,LPX_K_ITCNT) ;
     isIterationLimitReached_ = false ;
+		isTimeLimitReached_ = false ;
     isAbandoned_ = false ;
     isPrimInfeasible_ = false ;
     isDualInfeasible_ = false ;
@@ -366,8 +373,9 @@
       case LPX_E_NODFS:
       { isDualInfeasible_ = true ;
 	break ; }
+      case LPX_E_TMLIM:
+      { isTimeLimitReached_ = true ; } // no break
       case LPX_E_ITLIM:
-      case LPX_E_TMLIM:
       { isIterationLimitReached_ = true ;
 	break ; }
       case LPX_E_SING:
@@ -841,6 +849,11 @@
 	return isIterationLimitReached_;
 }
 
+bool OGSI::isTimeLimitReached() const
+{
+	return isTimeLimitReached_;
+}
+
 //#############################################################################
 // WarmStart related methods
 //#############################################################################
@@ -3139,6 +3152,7 @@
   hotStartRStatSize_ = 0;
 
   isIterationLimitReached_ = false;
+  isTimeLimitReached_ = false;
   isAbandoned_ = false;
   isPrimInfeasible_ = false;
   isDualInfeasible_ = false;
@@ -3252,6 +3266,7 @@
         iter_used_ = 0;
 	isAbandoned_ = false;
 	isIterationLimitReached_ = false;
+	isTimeLimitReached_ = false;
 	isPrimInfeasible_ = false;
 	isDualInfeasible_ = false;
 	delete [] colsol_;
Index: Osi/src/OsiGlpk/OsiGlpkSolverInterface.hpp
===================================================================
--- Osi/src/OsiGlpk/OsiGlpkSolverInterface.hpp	(revision 1023)
+++ Osi/src/OsiGlpk/OsiGlpkSolverInterface.hpp	(working copy)
@@ -17,6 +17,9 @@
 #include "CoinPackedMatrix.hpp"
 #include "CoinWarmStartBasis.hpp"
 
+// A define that tells whether methods to check for the timelimit exists in OsiGlpkSolverInterface
+#define OGSI_HAVE_TIMELIMIT 1
+
 /** GPLK Solver Interface
 
     Instantiation of OsiGlpkSolverInterface for GPLK
@@ -90,6 +93,8 @@
     virtual bool isDualObjectiveLimitReached() const;
     /// Iteration limit reached?
     virtual bool isIterationLimitReached() const;
+    /// Time limit reached?
+    virtual bool isTimeLimitReached() const;
   //@}
 
   //---------------------------------------------------------------------------
@@ -770,6 +775,8 @@
   // Status information
   /// glpk stopped on iteration limit
   bool isIterationLimitReached_;
+  /// glpk stopped on time limit
+  bool isTimeLimitReached_;
   /// glpk abandoned the problem 
   bool isAbandoned_;
   /*! \brief  glpk stopped on lower objective limit
