diff -Naur orig/glpk-4.15/examples/Makefile.in glpk-4.15/examples/Makefile.in
--- orig/glpk-4.15/examples/Makefile.in	2007-02-18 10:00:00.000000000 +0100
+++ glpk-4.15/examples/Makefile.in	2007-03-11 15:27:49.000000000 +0100
@@ -170,7 +170,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-INCLUDES = -I../include
+INCLUDES = -I$(srcdir)/../include
 LDADD = ../src/libglpk.la
 glpsol_SOURCES = glpsol.c
 tspsol_SOURCES = tspsol.c
diff -Naur orig/glpk-4.15/src/glplpx03.c glpk-4.15/src/glplpx03.c
--- orig/glpk-4.15/src/glplpx03.c	2007-02-18 10:00:00.000000000 +0100
+++ glpk-4.15/src/glplpx03.c	2007-03-11 15:26:36.000000000 +0100
@@ -305,7 +305,7 @@
             lp->relax = val;
             break;
          case LPX_K_TOLBND:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (!(DBL_EPSILON <= val))
                fault("lpx_set_real_parm: TOLBND = %g; invalid value",
                   val);
             if (lp->tol_bnd > val)
@@ -316,7 +316,7 @@
             lp->tol_bnd = val;
             break;
          case LPX_K_TOLDJ:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (!(DBL_EPSILON <= val))
                fault("lpx_set_real_parm: TOLDJ = %g; invalid value",
                   val);
             if (lp->tol_dj > val)
@@ -327,7 +327,7 @@
             lp->tol_dj = val;
             break;
          case LPX_K_TOLPIV:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (!(DBL_EPSILON <= val))
                fault("lpx_set_real_parm: TOLPIV = %g; invalid value",
                   val);
             lp->tol_piv = val;
@@ -345,13 +345,13 @@
             lp->out_dly = val;
             break;
          case LPX_K_TOLINT:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (!(DBL_EPSILON <= val))
                fault("lpx_set_real_parm: TOLINT = %g; invalid value",
                   val);
             lp->tol_int = val;
             break;
          case LPX_K_TOLOBJ:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (!(DBL_EPSILON <= val))
                fault("lpx_set_real_parm: TOLOBJ = %g; invalid value",
                   val);
             lp->tol_obj = val;
diff -Naur orig/glpk-4.15/src/Makefile.in glpk-4.15/src/Makefile.in
--- orig/glpk-4.15/src/Makefile.in	2007-02-18 10:00:00.000000000 +0100
+++ glpk-4.15/src/Makefile.in	2007-03-11 15:31:39.000000000 +0100
@@ -185,7 +185,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-INCLUDES = -I../include
+INCLUDES = -I$(srcdir)/../include
 lib_LTLIBRARIES = libglpk.la
 libglpk_la_LDFLAGS = -version-info 0:0:0
 libglpk_la_SOURCES = \
