diff -Naur orig/glpk-4.9/examples/Makefile.in glpk-4.9/examples/Makefile.in
--- orig/glpk-4.9/examples/Makefile.in	2006-01-17 10:00:00.000000000 +0100
+++ glpk-4.9/examples/Makefile.in	2006-12-18 10:06:46.000000000 +0100
@@ -63,7 +63,7 @@
 RANLIB = @RANLIB@
 VERSION = @VERSION@
 
-INCLUDES = -I../include
+INCLUDES = -I$(srcdir)/../include
 
 LDADD = ../src/libglpk.a
 
diff -Naur orig/glpk-4.9/src/glplpx3.c glpk-4.9/src/glplpx3.c
--- orig/glpk-4.9/src/glplpx3.c	2006-01-17 10:00:00.000000000 +0100
+++ glpk-4.9/src/glplpx3.c	2006-12-18 10:54:08.000000000 +0100
@@ -305,7 +305,7 @@
             lp->relax = val;
             break;
          case LPX_K_TOLBND:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (DBL_EPSILON >= val)
                fault("lpx_set_real_parm: TOLBND = %g; invalid value",
                   val);
             if (lp->tol_bnd > val)
@@ -316,7 +316,7 @@
             lp->tol_bnd = val;
             break;
          case LPX_K_TOLDJ:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (DBL_EPSILON >= val)
                fault("lpx_set_real_parm: TOLDJ = %g; invalid value",
                   val);
             if (lp->tol_dj > val)
@@ -327,7 +327,7 @@
             lp->tol_dj = val;
             break;
          case LPX_K_TOLPIV:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (DBL_EPSILON >= val)
                fault("lpx_set_real_parm: TOLPIV = %g; invalid value",
                   val);
             lp->tol_piv = val;
@@ -345,13 +345,13 @@
             lp->out_dly = val;
             break;
          case LPX_K_TOLINT:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (DBL_EPSILON >= val)
                fault("lpx_set_real_parm: TOLINT = %g; invalid value",
                   val);
             lp->tol_int = val;
             break;
          case LPX_K_TOLOBJ:
-            if (!(DBL_EPSILON <= val && val <= 0.001))
+            if (DBL_EPSILON >= val)
                fault("lpx_set_real_parm: TOLOBJ = %g; invalid value",
                   val);
             lp->tol_obj = val;
diff -Naur orig/glpk-4.9/src/Makefile.in glpk-4.9/src/Makefile.in
--- orig/glpk-4.9/src/Makefile.in	2006-01-17 10:00:00.000000000 +0100
+++ glpk-4.9/src/Makefile.in	2006-12-18 10:04:12.000000000 +0100
@@ -63,7 +63,7 @@
 RANLIB = @RANLIB@
 VERSION = @VERSION@
 
-INCLUDES = -I../include
+INCLUDES = -I$(srcdir)/../include
 
 lib_LIBRARIES = libglpk.a
 
